src/MetadataTools.jl

```diff
@@ -146,10 +176,9 @@ function get_upper_limit(pkg::PkgMeta)
         julia_max_ver = v"0.0.0"
         # Check if there is a Julia max version dependency
         for req in ver.requires
-            !contains(req,"julia") && continue
-            s = split(req," ")
-            length(s) != 3 && continue
-            julia_max_ver = convert(VersionNumber,s[3])
+            req.name!="julia" && continue
+            isnull(req.upper_bound) && continue
+            julia_max_ver = get(req.upper_bound)
```

Comment:
Ok, I've pushed something that handles the case with one interval and throws an error otherwise. Probably more robust than the old code, that might or might not have thrown an error.